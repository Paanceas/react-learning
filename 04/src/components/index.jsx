export * from './AddCategory';
export * from './GifGItem';
export * from './GifGrid';