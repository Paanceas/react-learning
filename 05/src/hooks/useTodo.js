import { useEffect, useReducer } from "react";
import { todoReducer } from "../08-useReducer/todoReducer";

const init = () => {
    return JSON.parse(localStorage.getItem('todos') || [])
}

export const useTodo = () => {

    const [ todos, dispatch ] = useReducer(todoReducer, [], init);

    const handleNewTodo = (todo) => {
      const action = {
        type: '[TODO] Add Todo',
        payload: todo
      }
      dispatch(action);
    }

    const handleDeleteTodo = (todo) => {
      dispatch({
        type: '[TODO] Remove Todo',
        payload: todo
      });
    }

    const handleToggleTodo = (todo) => {
        const action = {
          type: '[TODO] Toggle Todo',
          payload: todo
        }
        dispatch(action);
      }

    useEffect(() => {
      localStorage.setItem('todos', JSON.stringify(todos));
    }, [todos])

  return {
    todos,
    handleNewTodo,
    handleDeleteTodo,
    handleToggleTodo
  }
}
