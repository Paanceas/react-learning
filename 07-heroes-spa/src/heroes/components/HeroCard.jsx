import { Link } from "react-router-dom";


const CharacterByHero = ({alter_ego, characters}) => {
  if(alter_ego === characters) return(<></>);
  return (<p>{characters}</p>);
}

export const HeroCard = ({ hero }) => {

  const heroImg = `/assets/heroes/${hero.id}.jpg`;

  return (
    <div className="col ">
      <div className="card">
        <div className="row no-gutters">
          <div className="col-4">
            <img className="card-img" src={heroImg} alt={hero.superhero} />
          </div>
          <div className="col-8">
            <div className="card-body">
              <h5 className="card-title">{hero.superhero}</h5>
              <p className="card-text">{hero.alter_ego}</p>
              <CharacterByHero characters={hero.characters} alter_ego={hero.alter_ego} />
              <p className="card text">
                <small>{hero.first_appearance}</small>
              </p>
              <Link to={`/hero/${hero.id}`} >
                MÃ¡s...
              </Link>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
