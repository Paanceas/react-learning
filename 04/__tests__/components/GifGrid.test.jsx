import { render, screen } from "@testing-library/react";
import { GifGrid } from "../../src/components/GifGrid";
import { useFetchGifs } from "../../src/hooks/useFetchGifs";

jest.mock("../../src/hooks/useFetchGifs");

describe("Pruebas en GifGrid", () => {
  const props = {
    category: "One pice",
  };

  const initUseFetchGifs = {
    images: [],
    isLoading: true,
  };

  const endUseFetchGifs = {
    images: [
      {
        id: "ABC",
        title: "Saitama",
        url: "https://gogle.com.co/img.png",
      },
      {
        id: "ABCD",
        title: "Saitama",
        url: "https://gogle.com.co/img.png",
      },
    ],
    isLoading: false,
  };

  test("debe hacer match con el snapshot del componente", () => {
    useFetchGifs.mockReturnValue(initUseFetchGifs);
    const { container } = render(<GifGrid {...props} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar el loading inicial", () => {
    useFetchGifs.mockReturnValue(initUseFetchGifs);

    render(<GifGrid {...props} />);
    expect(screen.getByText("Cargando...")).toBeTruthy();
    expect(screen.getByText(props.category)).toBeTruthy();
  });

  test("Debe de mostrar items cuando se cargan las imagenes useFetchGifs", () => {
    useFetchGifs.mockReturnValue(endUseFetchGifs);
    render(<GifGrid {...props} />);
    expect(screen.getAllByRole("img").length).toBe(
      endUseFetchGifs.images.length
    );
  });
});
