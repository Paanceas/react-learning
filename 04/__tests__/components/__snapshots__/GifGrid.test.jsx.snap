// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid debe hacer match con el snapshot del componente 1`] = `
<div>
  <h3>
     
    One pice
     
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
